File: programmers/92334.js
// 신고 결과 받기

// 입력 예시
// const id_list = ['muzi','frodo','apeach', 'neo']
// const report = ['muzi frodo', 'apeach frodo', 'frodo neo', 'muzi neo', 'apeach muzi']
// const k = 2

// const id_list = ['con', 'ryan']
// const report = ["ryan con", "ryan con", "ryan con", "ryan con"]
// const k = 3

function solution(id_list, report, k) {
    // 유저 수만큼 답지 초기화
    var answer = [];
    var num_report = [];
    var blacklist = [];

    for (var i = 0; i < id_list.length; i++) {
        answer.push(0);
        num_report.push(0);
    }

    // report 배열에서 중복 요소를 제거
    report = [...new Set(report)]

    // 신고 누적 횟수를 담은 배열 만들기
    report.forEach(function (item) {
        var i = id_list.indexOf(item.split(' ')[1]);
        num_report[i] += 1;
    })
    // console.log(num_report)

    // 정지먹은 유저들의 이름이 담긴 배열 만들기
    num_report.forEach(function (item, index) {
        if (item >= k ) {
            blacklist.push(id_list[index]);
        }
    })
    // console.log(blacklist)

    // 정지먹은 유저들을 신고한 유저들에게 이메일 전송!
    report.forEach(function (item) {
        if (blacklist.includes(item.split(' ')[1])) {
            var bui = id_list.indexOf(item.split(' ')[0])
            answer[bui] += 1;
        }
    })
    // console.log(answer)

    return answer;
}

File: programmers/lvl_1/92334.py
# 신고 결과 받기


def solution(id_list, report, k):
    # 유저 수만큼 답지 초기화
    answer = []
    for i in range(len(id_list)):
        answer.append(0)

    # Remove Duplicates -> 같은 신고는 하나로 간주.
    report = list(dict.fromkeys(report))

    # 신고 누적 횟수를 담은 딕셔너리 만들기
    dic = {}
    for i in range(len(report)):
        if report[i].split()[-1] in dic.keys():
            dic[report[i].split()[-1]] += 1
            # 유저 신고 누적 횟수 증가 (UPDATE)
        else:
            dic[report[i].split()[-1]] = 1
            # 신고를 당한 유저 엔트리 생성 (CREATE)
    # print(dic)

    # 정지먹은 유저들의 이름이 담긴 리스트 만들기
    blacklist = []
    for key, val in dic.items():
        if val >= k:
            blacklist.append(key)
            # 그 value를 가진 key를 블랙리스트에 추가
    # print(blacklist)

    # 정지먹은 유저들을 신고한 유저들에게 이메일 전송!
    for i in range(len(report)):
        if report[i].split()[-1] in blacklist:
            answer[id_list.index(report[i].split()[0])] += 1

    return answer
